prime-router/src/test/kotlin/serializers/Hl7SerializerTests.kt

```diff
@@ -113,37 +115,68 @@ NTE|1|L|This is a final comment|RE"""
-            every { it.suppressHl7Fields }.returns(null)
-            every { it.useBlankInsteadOfUnknown }.returns(null)
-            every { it.convertTimestampToDateTime }.returns(null)
-            every { it.truncateHDNamespaceIds }.returns(false)
-            every { it.phoneNumberFormatting }.returns(Hl7Configuration.PhoneNumberFormatting.STANDARD)
-            every { it.usePid14ForPatientEmail }.returns(false)
-            every { it.reportingFacilityName }.returns(null)
-            every { it.reportingFacilityId }.returns(null)
-            every { it.reportingFacilityIdType }.returns(null)
-            every { it.cliaForOutOfStateTesting }.returns("1234FAKECLIA")
-            every { it.useOrderingFacilityName }.returns(Hl7Configuration.OrderingFacilityName.STANDARD)
-            every { it.cliaForSender }.returns(mapOf())
-        }
-        val receiver = mockkClass(Receiver::class).also {
-            every { it.translation }.returns(hl7Config)
-            every { it.format }.returns(Report.Format.HL7)
-            every { it.organizationName }.returns("ca-dph")
-        }
-
-        val testReport = csvSerializer.readExternal(schema, inputStream, listOf(TestSource), receiver).report ?: fail()
+        val hl7Config = Hl7Configuration(
+            messageProfileId = "",
+            receivingApplicationOID = "",
+            receivingApplicationName = "",
+            receivingFacilityName = "",
+            receivingFacilityOID = "",
+            receivingOrganization = "",
+            replaceValue = mapOf("PID-22-3" to "CDCREC,-,testCDCREC", "MSH-9" to "MSH-10"),
+            cliaForOutOfStateTesting = "1234FAKECLIA"
+        )
+        val receiver = Receiver("mock", "ca-phd", "covid-19", translation = hl7Config)
+        val pdiInput = csvSerializer.readExternal(schema, inputStream, listOf(TestSource), receiver).report ?: fail()
+        val testReport = translator.translateByReceiver(pdiInput, receiver)
         val output = serializer.createMessage(testReport, 2)
         assertThat(output).isNotNull()
     }
 
+    @Test
+    fun `test message with bad NPI`() {
+        val inputStream = File("./src/test/unit_test_files/fake-pdi-covid-19.csv").inputStream()
+        val schema = "primedatainput/pdi-covid-19"
+        val hl7Config = Hl7Configuration(
+            messageProfileId = "",
+            receivingApplicationOID = "",
+            receivingApplicationName = "",
+            receivingFacilityName = "",
+            receivingFacilityOID = "",
+            receivingOrganization = "",
+            suppressNonNPI = false,
+        )
+        val receiver = Receiver("mock", "ca-phd", "covid-19", translation = hl7Config)
+        val pdiInput = csvSerializer.readExternal(schema, inputStream, listOf(TestSource), receiver).report ?: fail()
+        val testReport = translator.translateByReceiver(pdiInput, receiver)
+        val output = serializer.buildMessage(testReport, 2)
```

Comment:
This is a new test pattern that inspects the serialized HL7 for the expected values.